### ISSUE TITLE
Fit method printout is broken in version 2.1 

### ISSUE THREAD

**Issue by OmerLiberman** [tensorflow/tensorflow#36332]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): MACOS
- TensorFlow installed from (source or binary): source
- TensorFlow version (use command below): 2.1.0
- Python version: 3.7.4

** Info **
The out-prints of in the fit method in version 2.0:
<img width="1440" alt="Screen Shot 2020-01-30 at 11 08 16" src="https://user-images.githubusercontent.com/38648402/73436454-36441700-4353-11ea-86e1-7ff0023ae0b1.png">

The out-prints of in the fit method in version 2.1:
<img width="1440" alt="Screen Shot 2020-01-30 at 11 11 07" src="https://user-images.githubusercontent.com/38648402/73436538-5d024d80-4353-11ea-8fcb-d40560a2f79d.png">

Watch the output of the first epoch (the line, not the values), it prints Epcoh1/3 at the end of the line.

**Comment by ravikyram** [comment]
@OmerLiberman 

Can you please provide the simple standalone code to reproduce the issue in our environment. It helps us in localizing the issue faster. Thanks!

**Comment by ravikyram** [comment]
@OmerLiberman 

Any update on this issue please. Thanks!

**Comment by OmerLiberman** [comment]
I don't know what changed but it seems fixed

**Comment by OmerLiberman** [comment]
The output is:
`

13m 43s | 220 | Epoch 1/3
-- | -- | --
13m 43s | 221 | 2020-02-13 08:01:22.217883: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcublas.so.10
13m 43s | 222 | 2020-02-13 08:01:22.788927: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudnn.so.7
15m 55s | 223 | 1/82 [..............................] - ETA: 13:15 - loss: 4.1689 - accuracy: 0.4919 2/82 [..............................] - ETA: 9:49 - loss: 9.9026 - accuracy: 0.4879  3/82 [>.............................] - ETA: 7:48 - loss: 8.2984 - accuracy: 0.4960 4/82 [>.............................] - ETA: 6:46 - loss: 6.5161 - accuracy: 0.5544 5/82 [>.............................] - ETA: 5:59 - loss: 5.7836 - accuracy: 0.5597 6/82 [=>............................] - ETA: 5:16 - loss: 5.0436 - accuracy: 0.5847 7/82 [=>............................] - ETA: 4:33 - loss: 4.3812 - accuracy: 0.6244 8/82 [=>............................] - ETA: 4:10 - loss: 3.9348 - accuracy: 0.6462 9/82 [==>...........................] - ETA: 3:51 - loss: 3.6170 - accuracy: 0.6541 10/82 [==>...........................] - ETA: 3:36 - loss: 3.3158 - accuracy: 0.6702 11/82 [===>..........................] - ETA: 3:23 - loss: 3.0370 - accuracy: 0.6906 12/82 [===>..........................] - ETA: 3:13 - loss: 2.8060 - accuracy: 0.7077 13/82 [===>..........................] - ETA: 3:03 - loss: 2.6200 - accuracy: 0.7205 14/82 [====>.........................] - ETA: 2:55 - loss: 2.4594 - accuracy: 0.7327 15/82 [====>.........................] - ETA: 2:48 - loss: 2.3163 - accuracy: 0.7430 16/82 [====>.........................] - ETA: 2:41 - loss: 2.1798 - accuracy: 0.7558 17/82 [=====>........................] - ETA: 2:35 - loss: 2.0598 - accuracy: 0.7673 18/82 [=====>........................] - ETA: 2:30 - loss: 1.9513 - accuracy: 0.7780 19/82 [=====>........................] - ETA: 2:25 - loss: 1.8566 - accuracy: 0.7867 20/82 [======>.......................] - ETA: 2:20 - loss: 1.7731 - accuracy: 0.7944 21/82 [======>.......................] - ETA: 2:16 - loss: 1.6957 - accuracy: 0.8015 22/82 [=======>......................] - ETA: 2:11 - loss: 1.6245 - accuracy: 0.8090 23/82 [=======>......................] - ETA: 2:07 - loss: 1.5569 - accuracy: 0.8161 24/82 [=======>......................] - ETA: 2:04 - loss: 1.4935 - accuracy: 0.8233 25/82 [========>.....................] - ETA: 2:00 - loss: 1.4365 - accuracy: 0.8298 26/82 [========>.....................] - ETA: 1:57 - loss: 1.3848 - accuracy: 0.8353 27/82 [========>.....................] - ETA: 1:53 - loss: 1.3362 - accuracy: 0.8407 28/82 [=========>....................] - ETA: 1:50 - loss: 1.2914 - accuracy: 0.8453 29/82 [=========>....................] - ETA: 1:47 - loss: 1.2492 - accuracy: 0.8497 30/82 [=========>....................] - ETA: 1:44 - loss: 1.2095 - accuracy: 0.8540 31/82 [==========>...................] - ETA: 1:41 - loss: 1.1723 - accuracy: 0.8582 32/82 [==========>...................] - ETA: 1:39 - loss: 1.1365 - accuracy: 0.8624 33/82 [===========>..................] - ETA: 1:36 - loss: 1.1026 - accuracy: 0.8664 34/82 [===========>..................] - ETA: 1:33 - loss: 1.0706 - accuracy: 0.8703 35/82 [===========>..................] - ETA: 1:31 - loss: 1.0406 - accuracy: 0.8737 36/82 [============>.................] - ETA: 1:28 - loss: 1.0124 - accuracy: 0.8770 37/82 [============>.................] - ETA: 1:26 - loss: 0.9857 - accuracy: 0.8800 38/82 [============>.................] - ETA: 1:24 - loss: 0.9603 - accuracy: 0.8829 39/82 [=============>................] - ETA: 1:21 - loss: 0.9367 - accuracy: 0.8854 40/82 [=============>................] - ETA: 1:19 - loss: 0.9137 - accuracy: 0.8882 41/82 [==============>...............] - ETA: 1:17 - loss: 0.8918 - accuracy: 0.8907 42/82 [==============>...............] - ETA: 1:14 - loss: 0.8708 - accuracy: 0.8933 43/82 [==============>...............] - ETA: 1:12 - loss: 0.8510 - accuracy: 0.8956 44/82 [===============>..............] - ETA: 1:10 - loss: 0.8318 - accuracy: 0.8980 45/82 [===============>..............] - ETA: 1:08 - loss: 0.8134 - accuracy: 0.9002 46/82 [===============>..............] - ETA: 1:06 - loss: 0.7960 - accuracy: 0.9023 47/82 [================>.............] - ETA: 1:04 - loss: 0.7793 - accuracy: 0.9043 48/82 [================>.............] - ETA: 1:02 - loss: 0.7634 - accuracy: 0.9061 49/82 [================>.............] - ETA: 1:00 - loss: 0.7479 - accuracy: 0.9080 50/82 [=================>............] - ETA: 58s - loss: 0.7331 - accuracy: 0.9098  51/82 [=================>............] - ETA: 56s - loss: 0.7188 - accuracy: 0.9116 52/82 [==================>...........] - ETA: 54s - loss: 0.7051 - accuracy: 0.9132 53/82 [==================>...........] - ETA: 52s - loss: 0.6919 - accuracy: 0.9149 54/82 [==================>...........] - ETA: 50s - loss: 0.6792 - accuracy: 0.9164 55/82 [===================>..........] - ETA: 48s - loss: 0.6669 - accuracy: 0.9180 56/82 [===================>..........] - ETA: 46s - loss: 0.6550 - accuracy: 0.9194 57/82 [===================>..........] - ETA: 44s - loss: 0.6436 - accuracy: 0.9208 58/82 [====================>.........] - ETA: 42s - loss: 0.6326 - accuracy: 0.9222 59/82 [====================>.........] - ETA: 40s - loss: 0.6219 - accuracy: 0.9235 60/82 [====================>.........] - ETA: 38s - loss: 0.6116 - accuracy: 0.9248 61/82 [=====================>........] - ETA: 37s - loss: 0.6016 - accuracy: 0.9260 62/82 [=====================>........] - ETA: 35s - loss: 0.5919 - accuracy: 0.9272 63/82 [======================>.......] - ETA: 33s - loss: 0.5825 - accuracy: 0.9284 64/82 [======================>.......] - ETA: 31s - loss: 0.5735 - accuracy: 0.9295 65/82 [======================>.......] - ETA: 29s - loss: 0.5647 - accuracy: 0.9306 66/82 [=======================>......] - ETA: 27s - loss: 0.5561 - accuracy: 0.9316 67/82 [=======================>......] - ETA: 26s - loss: 0.5478 - accuracy: 0.9327 68/82 [=======================>......] - ETA: 24s - loss: 0.5398 - accuracy: 0.9336 69/82 [========================>.....] - ETA: 22s - loss: 0.5320 - accuracy: 0.9346 70/82 [========================>.....] - ETA: 20s - loss: 0.5245 - accuracy: 0.9355 71/82 [========================>.....] - ETA: 19s - loss: 0.5171 - accuracy: 0.9364 72/82 [=========================>....] - ETA: 17s - loss: 0.5099 - accuracy: 0.9373 73/82 [=========================>....] - ETA: 15s - loss: 0.5030 - accuracy: 0.9382 74/82 [==========================>...] - ETA: 13s - loss: 0.4962 - accuracy: 0.9390 75/82 [==========================>...] - ETA: 12s - loss: 0.4896 - accuracy: 0.9398 76/82 [==========================>...] - ETA: 10s - loss: 0.4831 - accuracy: 0.9406 77/82 [===========================>..] - ETA: 8s - loss: 0.4769 - accuracy: 0.9414  78/82 [===========================>..] - ETA: 6s - loss: 0.4708 - accuracy: 0.9422 79/82 [===========================>..] - ETA: 5s - loss: 0.4648 - accuracy: 0.9429 80/82 [============================>.] - ETA: 3s - loss: 0.4591 - accuracy: 0.9436 81/82 [============================>.] - ETA: 1s - loss: 0.4534 - accuracy: 0.9443Epoch 1/3
16m 5s | 224 | 1/82 [..............................] - ETA: 2:30 - loss: 0.1519 - accuracy: 0.9516 2/82 [..............................] - ETA: 1:19 - loss: 0.1519 - accuracy: 0.9516 3/82 [>.............................] - ETA: 1:02 - loss: 0.1519 - accuracy: 0.9516 4/82 [>.............................] - ETA: 54s - loss: 0.1519 - accuracy: 0.9516  5/82 [>.............................] - ETA: 48s - loss: 0.1519 - accuracy: 0.9516 6/82 [=>............................] - ETA: 45s - loss: 0.1519 - accuracy: 0.9516 7/82 [=>............................] - ETA: 42s - loss: 0.1519 - accuracy: 0.9516 8/82 [=>............................] - ETA: 40s - loss: 0.1519 - accuracy: 0.9516 9/82 [==>...........................] - ETA: 38s - loss: 0.1519 - accuracy: 0.9516 10/82 [==>...........................] - ETA: 37s - loss: 0.1519 - accuracy: 0.9516 11/82 [===>..........................] - ETA: 36s - loss: 0.1519 - accuracy: 0.9516 12/82 [===>..........................] - ETA: 34s - loss: 0.1519 - accuracy: 0.9516 13/82 [===>..........................] - ETA: 33s - loss: 0.1519 - accuracy: 0.9516 14/82 [====>.........................] - ETA: 33s - loss: 0.1519 - accuracy: 0.9516 15/82 [====>.........................] - ETA: 32s - loss: 0.1519 - accuracy: 0.9516 16/82 [====>.........................] - ETA: 31s - loss: 0.1519 - accuracy: 0.9516 17/82 [=====>........................] - ETA: 30s - loss: 0.1519 - accuracy: 0.9516 18/82 [=====>........................] - ETA: 29s - loss: 0.1519 - accuracy: 0.9516 19/82 [=====>........................] - ETA: 29s - loss: 0.1519 - accuracy: 0.9516 20/82 [======>.......................] - ETA: 28s - loss: 0.1519 - accuracy: 0.9516 82/82 [==============================] - 148s 2s/step - loss: 0.4479 - accuracy: 0.9450 - val_loss: 0.1519 - val_accuracy: 0.9516
16m 5s | 225 | Epoch 2/3

`

Watch the "Epoch .." printed out.

**Comment by OmerLiberman** [comment]
**The code I use:**

**for creating the image generator**
`
def __load_train_and_val_generators(data, image_size, val_size, image_color, batch_size, class_mode, classes):
	"""
	Returns two generators contains the files in the directory data..
	"""
	data_gen = tf.keras.preprocessing.image.ImageDataGenerator(validation_split=val_size)

	train_gen = data_gen.flow_from_directory(
		data,
		color_mode=image_color,
		target_size=image_size,
		batch_size=batch_size,
		class_mode=class_mode,
		subset='training',
		classes=classes,
		shuffle=True)

	val_gen = data_gen.flow_from_directory(
		data,
		color_mode=image_color,
		target_size=image_size,
		batch_size=batch_size,
		class_mode=class_mode,
		subset='validation',
		classes=classes,
		shuffle=True)

	return train_gen, val_gen
`

for running the training:
`
	def __train(self):
		train_generator, val_generator = load_generator(self.__arguments.data, self.__shape,
														self.__arguments.test_size, self.__arguments.image_color,
														self.__arguments.batch_size)
		steps_per_epoch_training = train_generator.n // self.__arguments.epochs
		steps_per_epoch_validation = val_generator.n // self.__arguments.epochs

		start_time = time.time()
		print("---start training---")
		self.__model.fit(train_generator,
						epochs=self.__arguments.epochs,
						workers=TensorflowTrainer.WORKERS,
						verbose=TensorflowTrainer.VERBOSE,
						steps_per_epoch=steps_per_epoch_training,
						validation_data=val_generator,
						validation_steps=steps_per_epoch_validation)
		print("---End training---")
		training_time = time.strftime("%H:%M:%S", time.gmtime(time.time() - start_time))
		self.__metrics['training_time'] = training_time
`

Tensorflow version in use : 2.1.0

**Comment by OmerLiberman** [comment]
@ravikyram 
sorry for the delay, 
finals at school.

**Comment by ravikyram** [comment]
@OmerLiberman 

Will it be possible to share colab link or simple standalone code with  proper indentation to reproduce the issue in our environment. It helps in localizing the issue faster. Thanks!

**Comment by OmerLiberman** [comment]
@ravikyram  
take this code : https://github.com/OmerLiberman/tf_2.0.1

run the file '''resnet50.py''' with '''--data=PATH" 
where PATH is a path to a local directory with with images splitted to single class in each sub-directory.

**Comment by OmerLiberman** [comment]
I am trying to debug it. 
It is in the method: 
```
def model_iteration(...):
```

which is in :
```
tensorflow_core/python/keras/engine/training_generator.py

**Comment by OmerLiberman** [comment]
check the methods:

```
def on_epoch_end(...)
```

in ```tensorflow.python.keras.callbacks.py```

**Comment by OmerLiberman** [comment]
check:

In:  ```tensorflow_core/python/keras/engine/training_generator.py```
row num 293: ```batch_logs = cbks.make_logs(model, batch_logs, batch_outs, mode)```

It happens in the last epoch

**Comment by OmerLiberman** [comment]
Have you fixed the bug?

**Comment by OmerLiberman** [comment]
@ravikyram 
<img width="1440" alt="Screen Shot 2020-03-05 at 10 21 52" src="https://user-images.githubusercontent.com/38648402/75961861-6ee38d00-5ecb-11ea-833b-145338b85646.png">

**Comment by OmerLiberman** [comment]
@ravikyram 
![a](https://user-images.githubusercontent.com/38648402/75961905-8589e400-5ecb-11ea-9a4c-a6f2981b6357.png)

**Comment by OmerLiberman** [comment]
problem was found: 
<img width="1440" alt="Screen Shot 2020-03-05 at 11 44 35" src="https://user-images.githubusercontent.com/38648402/75968871-c2a7a380-5ed6-11ea-824e-1071aae460b6.png">

it enters this line at the end of the epoch

**Comment by gowthamkpr** [comment]
Can you please reproduce it and share it on [colab](temp_weights = [layer.get_weights() for layer in model.layers]) as I am not able to find your gist @OmerLiberman. Thanks!

**Comment by mihaimaruseac** [comment]
@OmerLiberman please use ` ``` `, not ` ` ` around the code blocks (and definitely not images)

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by ravikyram** [comment]
@OmerLiberman 

Any update on this issue please. Thanks!

**Comment by gowthamkpr** [comment]
Tried to reproduce this issue using tensorflow 2.2.0rc2 but I am not running into any issues as mentioned above. Please find the github [gist](https://colab.sandbox.google.com/github/tensorflow/docs/blob/master/site/en/guide/keras/overview.ipynb#scrollTo=I7BcMHkB0E2U). Thanks!

**Comment by gowthamkpr** [comment]
Closing this issue as it has been inactive for more than 3 weeks. Please add additional comments for us to open this issue again. Thanks!
